### DISCUSSION TITLE
Do not use git branch when cwd is not a git repository

### DISCUSSION THREAD

**Discussion by dhruvmanila** [olimorris/persisted.nvim#73] [Ideas]
I think instead of using the default branch value, it would be better to save the session without the git branch info even if `use_git_branch` config option is true. This better reflects the actual state of the session and avoids creating invalid state.

**Comment by olimorris** [comment]
Yep I totally agree. This is a legacy effect of when a user creates a session without doing `git init`, then eventually makes it a git repo and doesn't want to lose the session. This was a really simple approach to solving this.

Happy to accept any PR that work within those constraints.
